### ISSUE TITLE
Declare inout quantum operations

### ISSUE THREAD

**Issue by mark-koch** [CQCL/guppylang#378]

**Comment by mark-koch** [comment]
Once #413 and #414 are done, we need to declare owned quantum functions instead

**Comment by ss2165** [comment]
Closing as obsolete, see #465

**Comment by croyzor** [comment]
I don't think this is obsolete. The quickest path forward (as in #486) is to keep using the existing quantum library in tests and rename it to `quantum_functional`. The `inout` quantum ops which would live in `guppylang.prelude.quantum` need to be written with a quantum op compiler that treats them as `inout`, instead of the default.
